lib/common/common/src/bitpacking.rs

```diff
@@ -0,0 +1,138 @@
+pub struct BitPacker<'a> {
+    output: &'a mut Vec<u8>,
+    buf: u64,
+    buf_bits: u8,
+}
+
+impl<'a> BitPacker<'a> {
+    #[inline]
+    pub fn new(output: &'a mut Vec<u8>) -> Self {
+        output.clear();
+        Self {
+            output,
+            buf: 0,
+            buf_bits: 0,
+        }
+    }
+
+    #[inline]
+    pub fn push(&mut self, value: u32, bits: u8) {
```

Comment:
Could you please provide some examples of how to use this function in real-life scenario. For instance, where `bits` is coming from? I see in the provided tests, that bits are used to generate value. Does it means we can derive `bits` from `value` inside the push function?